.ci/README.md

```diff
@@ -1,63 +1,87 @@
 
-## `create-prs`
+<!--TODO(ndmckinley) Make these commands resilient to missing files and require no setup.-->
 
-The third stage's goal is to create the downstream PRs.
+```
+fly execute -t main --config .ci/magic-modules/branch.yml --input magic-modules=. --output magic-modules-branched=/tmp/magic-modules-branched/ --include-ignored
+fly execute -t main --config .ci/magic-modules/generate-terraform.yml --input magic-modules-branched=/tmp/magic-modules-branched  --output terraform-generated=/tmp/terraform-output/  --include-ignored
+```
 
-It takes three inputs:
-* The CI repository
-* A copy of `magic-modules` which has passed the test stage
-* The initial PR
+This tells Concourse to upload the current directory (including files that `git` treats as irrelevant, like the `.git` directory).  It treats the repo as if it were the PR that the Magician is running against, creates a branch, and then generates the terraform repository, placing it in /tmp/terraform-output.
 
-It simply runs `magic-modules/create-pr.yml`, which creates the downstream pull requests using the `hub` CLI.  When complete, it updates `magic-modules` to point to the new submodule commits (for added clarity on the pull request), comments on that PR with a list of downstream PRs, and marks it as successfully generated.  If it fails, it marks generation as failed.
+**If you want to run the terraform tests on the generated code**, you should run this command:
 
-## `merge-prs`
+```
+fly execute -t main --config .ci/unit-tests/test-terraform.yml --input terraform=/tmp/terraform-output  --include-ignored
+```
 
-The fourth stage's goal is to merge `magic-modules` PRs after they have been approved and all the downstream PRs have been merged.
+**If you want to run the terraform tests on the existing submodule**, you should run this command:
 
-### `merge-and-update`
+```
+fly execute -t main --config .ci/unit-tests/task.yml --input magic-modules=. --include-ignored
+```
 
-`merge-and-update` takes two inputs:
-* The approved PR, after all downstream PRs have been merged.
-* This CI repository.
+# CI For MagicModules Developers
+If you develop MagicModules (code generation features), here are the things you'll want to know.
 
-It then runs `magic-modules/merge.yml`, which declares one output:
+## Adding Tests
+You can easily add things to the test suites.  The version of the `.yml` and `.sh` files in this subdirectory which are run when you call `fly execute` are the versions at HEAD in your local copy of the repository.  If you write new tests which require some setup, you can just add them to the shell scripts which are already being executed - if the overall shell script exits with a nonzero code, the task will be marked as failing.
 
-* The `magic-modules` PR repo after it has been updated to be ready to merge.
+# CI For CI Developers
```

Comment:
Consider moving this inside .ci/README.md and just link to it from here.

Reply:
That's this file.  :)